cmake/FindMKL.cmake

```diff
@@ -0,0 +1,86 @@
+###########################################################
+# Adapted from PRACE course materials at:
+# https://repository.prace-ri.eu/git/CodeVault/hpc-kernels/spectral_methods/-/blob/master/cmake/Modules/FindMKL.cmake
+###########################################################
+
+# This script looks for MKL in two locations:
+# - The environment variable MKLROOT, which is defined by
+#    sourcing an MKL environment
+# - The directory `/opt/intel/mkl`, which is a common
+#    install location for MKL.
+# It may be possible to install MKL using python -m pip,
+# though this is not guaranteed to be found, and may
+# require explicitly setting the MKLROOT variable.
+
+###########################################################
+# Stage 1: find the root directory
+###########################################################
+
+set(MKLROOT_PATH $ENV{MKLROOT})
+if (NOT MKLROOT_PATH)
+    if ($ENV{ONEAPI_ROOT})
+        set(MKLROOT_PATH $ENV{ONEAPI_ROOT}/mkl/latest)
+    endif ()
+endif ()
+if (NOT MKLROOT_PATH)
+    foreach(P IN ITEMS "/opt/intel/oneapi/mkl/latest" "/opt/intel/mkl")
+        if (EXISTS "${P}")
+            set(MKLROOT_PATH "${P}")
+            message(STATUS "Found MKL: ${P}")
+            break()
+        endif ()
+  endforeach()
+endif ()
```

Comment:
The indentation spacing is inconsistent. Could you please fix that?

Reply:
Never let an optician say you don't have good eyes, sir!

Reply:
Looks good now!